static/js/composebox_typeahead.js

```diff
@@ -841,6 +857,54 @@ exports.content_typeahead_selected = function (item, event) {
         // with the topic and the final **.
         const start = beginning.length - this.token.length;
         beginning = beginning.substring(0, start) + item + '** ';
+    } else if (this.completing === 'time_jump') {
+        const flatpickr_input = $("<input id='#timestamp_flatpickr'>");
+        let timeobject;
+        let timestring = beginning.substring(beginning.lastIndexOf('!time'));
+        if (timestring.startsWith('!time(') && timestring.endsWith(')')) {
+            timestring = timestring.substring(6, timestring.length - 1);
+            moment.suppressDeprecationWarnings = true;
+            try {
+                timeobject = moment(timestring).toDate();
+            } catch {
+                // do nothing. We'll just default to showing current date for the edit.
+            }
+        }
+
+        const instance = flatpickr_input.flatpickr({
+            mode: 'single',
+            enableTime: true,
+            clickOpens: false,
+            defaultDate: timeobject || moment().format(),
+            plugins: [new confirmDatePlugin({})], // eslint-disable-line new-cap, no-undef
+            positionElement: this.$element[0],
+            dateFormat: 'Z',
+            formatDate: (date) => {
+                const dt = moment(date);
+                return dt.local().format();
+            },
+        });
+        const container = $($(instance.innerContainer).parent());
+        container.on('click', '.flatpickr-calendar', (e) => {
+            e.stopPropagation();
+            e.preventDefault();
+        });
+
+        container.on('click', '.flatpickr-confirm', () => {
+            const datestr = flatpickr_input.val();
+            beginning = beginning.substring(0, beginning.lastIndexOf('!time')) +  `!time(${datestr}) `;
+            if (rest.startsWith(')')) {
+                rest = rest.slice(1);
+            }
+            textbox.val(beginning + rest);
+            textbox.caret(beginning.length, beginning.length);
```

Comment:
This feels a bit messy; could we do something more readable with more `compose_ui` functions?